scripts/infra/nvidia-setup.sh

```diff
@@ -47,12 +50,16 @@ FILEEOF
 
 rm -rf yum-packaging-precompiled-kmod
 dnf install libicu podman skopeo git rpm-build make openssl elfutils-libelf-devel python3.11 python3.11-devel -y
-if [ "${KERNEL_VERSION}" == "" ]; then \
-      RELEASE=$(dnf info --installed kernel-core | sort | awk -F: '/^Release/{print $2}' | tr -d '[:blank:]' | tail -n 1) \
-      && VERSION=$(dnf info --installed kernel-core | sort | awk -F: '/^Version/{print $2}' | tr -d '[:blank:]' | tail -n 1) \
-      && export KERNEL_VERSION="${VERSION}-${RELEASE}" ;\
-fi \
-    && dnf install -y "kernel-devel-${KERNEL_VERSION}" \
+if [ "${KERNEL_VERSION}" == "" ]; then
+      KERNELS=$(rpm -q --qf="%{BUILDTIME} %{VERSION}-%{RELEASE}\n" kernel-core | sort -n)
+      KERNEL_COUNT=$(echo "${KERNELS}" | wc -l)
+      LATEST_KERNEL=$(echo "${KERNELS}" | tail -1)
+      export KERNEL_VERSION=${LATEST_KERNEL#* }
```

Comment:
(No action required) I suspect export is not needed but it was there before, so it's ok.

Reply:
You're probably right. This PR is growing, and I'd prefer to merge this and address this in a separate PR.